### DISCUSSION TITLE
Can the "element-android" project be utilized as an Android Library?

### DISCUSSION THREAD

**Discussion by maksudur-rahman** [element-hq/element-meta#2412] [Ideas]
Is it feasible to use the "element-android" project as an Android Library? My app is a multimodule project that currently lacks any Hilt dependency. After adding VectorApplication.kt and SingletonModule.kt from the "vector-app" module to my project and designating the entry point as my app's MainActivity, an error labeled [Dagger/MissingBinding] occurs. What measures should be taken to address this problem?

**Comment by tkr-sh** [comment]
you should probably post it on https://github.com/element-hq/element-x-android

**Reply by awtj8o81ryywg793** [reply]
I know that I'm a _little_ late to the party here, but @maksudur-rahman isn't talking about [Element X](https://github.com/element-hq/element-x-android); rather, he's referring to [Element Classic](https://github.com/element-hq/element-android).

If you're still looking for support on this, @maksudur-rahman, you'll most likely be better served by asking the question in either the [Matrix Development](https://matrix.to/#/%23matrix-dev:matrix.org) room or the [Element Android](https://matrix.to/#/#element-android:matrix.org) room.

**Comment by awtj8o81ryywg793** [comment]
I'm _slightly_ late in pointing this out, but ideally, this discussion piece would have been filed under "Q&A" rather than "Ideas."

In terms of seeking further support, if it's still required, see my previous [comment](https://github.com/element-hq/element-meta/discussions/2412#discussioncomment-14662856).
